documentation/library_guide/api_for_dpcpp_kernels/random.rst

```diff
@@ -90,7 +90,7 @@ Random number generation is available for |dpcpp_long| device-side and host-side
 
     #include <iostream>
     #include <vector>
-    #include <CL/sycl.hpp>
```

Comment:
Waiting for final decision on this one. I'd like to point out that this is not just a matter of taste; for any other compliant SYCL compiler apart from DPC++ users following the example may get compiler errors if `CL/sycl.hpp` is used.

Reply:
Yes, "DPC++ compiler and libraries" does not contain <sycl/sycl.hpp> yet.  
Can we distinguish SYCL compiler apart from DPC++ by a compiler version macro?
In that case we can write something like that:
```
#if defined(__INTEL_LLVM_COMPILER) && SYCL_LANGUAGE_VERSION < "version_value"
#include <CL/sycl.hpp>
#else
#include <sycl/sycl.hpp>
#endif
```

Reply:
Regarding the examples. I think we may(or even must)  omit `<sycl/sycl.hpp>` in the examples and other user's code  which uses OneDPL, because the SYCL main header should be included once, within OneDPL library. For example,  here https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h#L24
So, when a user includes any  OneDPL public header, he gets SYCL library inclusion implicitly.

Reply:
@MikeDvorskiy, GitHub version of DPC++ has both ```<sycl/sycl.hpp>``` and ```<CL/sycl.hpp>``` while DPC++ supplied as a part of Intel® oneAPI toolkits has only ```<CL/sycl.hpp>```. 
What is benefit of using:
```c++
#if defined(__INTEL_LLVM_COMPILER) && SYCL_LANGUAGE_VERSION < "version_value"
#include <CL/sycl.hpp>
#else
#include <sycl/sycl.hpp>
#endif
```
instead of:
```c++
#if __has_include(<sycl/sycl.hpp>)
#include <sycl/sycl.hpp>
#else
#include <CL/sycl.hpp>
#endif
```
?

Meanwhile, the second approach does not require specifying version of the macro and its uplifting if necessary.

Reply:
@MikeDvorskiy, GitHub version of DPC++ has both ```<sycl/sycl.hpp>``` and ```<CL/sycl.hpp>``` while DPC++ supplied as a part of Intel® oneAPI toolkits has only ```<CL/sycl.hpp>```. 
What is benefit of using:
```c++
#if defined(__INTEL_LLVM_COMPILER) && SYCL_LANGUAGE_VERSION < "version_value"
#include <CL/sycl.hpp>
#else
#include <sycl/sycl.hpp>
#endif
```
instead of:
```c++
#if __has_include(<sycl/sycl.hpp>)
#include <sycl/sycl.hpp>
#else
#include <CL/sycl.hpp>
#endif
```
?

Meanwhile, the second approach does not require specifying version of the macro and its uplifting if necessary.

Reply:
I discussed the thread with @MikeDvorskiy offline. We agreed on the following:
- Get rid of inclusion of ```sycl.hpp``` in the examples and the guide.
- Use the approach with ```__has_include``` in ```sycl_defs.h```.